src/test/scala/com/yotpo/metorikku/code/steps/test/SelectiveMergeTests.scala

```diff
@@ -0,0 +1,170 @@
+package com.yotpo.metorikku.code.steps.test
+
+import com.yotpo.metorikku.code.steps.SelectiveMerge.merge
+import org.apache.log4j.{LogManager, Logger}
+import org.apache.spark.sql.types.StructField
+import org.apache.spark.sql.{DataFrame, SQLContext, SparkSession}
+import org.scalatest.{FunSuite, _}
+
+import scala.collection.mutable.ArrayBuffer
+
+//noinspection ScalaStyle
+class SelectiveMergeTests extends FunSuite with BeforeAndAfterEach {
+  private val log: Logger = LogManager.getLogger(this.getClass)
+  private var sparkSession : SparkSession = _
+
+  override def beforeEach() {
+    sparkSession = SparkSession.builder().appName("udf tests")
+      .master("local")
+      .config("", "")
+      .getOrCreate()
+  }
+
+  def assertSuccess(df1: DataFrame, df2: DataFrame, isEqual: Boolean): Unit = {
+    val sortedSchemeArrBuff: ArrayBuffer[String] = ArrayBuffer[String]()
+    df1.schema.sortBy({f: StructField => f.name}).map({f: StructField => sortedSchemeArrBuff += f.name})
+    val sortedSchemeArr: Array[String] = sortedSchemeArrBuff.sortWith(_<_).toArray
+
+
+    val sortedMergedDf = df1.orderBy("employee_name").select("employee_name", sortedSchemeArr:_*)
+    val sortedExpectedDf = df2.orderBy("employee_name").select("employee_name", sortedSchemeArr:_*)
+    val equals = sortedMergedDf.except(sortedExpectedDf).isEmpty
+
+    if (equals != isEqual) {
+      if (!equals) {
+        log.error("Actual and expected differ:")
+        log.error("Actual:")
+        sortedMergedDf.show()
+        log.error("Expected:")
+        sortedExpectedDf.show()
+        assert(false)
+      }
+      else {
+        log.error("Actual and expected are equal (but expected to differ)")
+        assert(false)
+      }
+    }
+  }
+
+  test("Selective merge") {
+    val employeeData1 = Seq(
+      ("James", 1, 11, 111, 1111),
+      ("Maria", 2, 22, 222, 2222)
+    )
```

Comment:
I think it's better to add here `.toDF("employee_name", "salary", "age", "fake", "fake2")`
I think it's more readable that way. WDYT?

Reply:
Done